src/async_wrap.cc

```diff
@@ -670,6 +673,9 @@ void AsyncWrap::AsyncReset(Local<Object> resource, double execution_async_id,
                                                      : execution_async_id;
   trigger_async_id_ = env()->get_default_trigger_async_id();
 
+  auto isolate = Isolate::GetCurrent();
+  resource_.Reset(isolate, resource);
```

Comment:
I think `resource_` should be cleared in `AsyncWrap::EmitDestroy`.
For resources like `HTTPParser` which stay alive forever this causes the last used wrapper to stay alive forever which is not intended (and causes test-gc-http-client.js to fail).